### ISSUE TITLE
"Maximum iterations in RReact" error with amanzi-tpls-0.97.3

### ISSUE THREAD

**Issue by zexuanxu** [amanzi/amanzi#449]
Recently when I'm running both amanzi-master and ats-modif4chemistry with alquimia-pflotran using the amanzi-tpls-0.97.3(4), the simulation failed with "Maximum iterations in RReact: stop:          501" error msg, which seems to be a pflotran issue. This issue does not occur with amanzi-0.88 (which used amanzi-tpls-0.95.7), or the older version of amanzi-0.87 or ats-modif4chemistry with amanzi-tpls-0.94.12. So it seems something in the newest tpls version (from 0.95.7 to 0.97.3) may cause the problem. Note that it also does not occur in all my cases (e.g., one of my 2D hillslope case seems working well with the same reactive network pflotran *.in file is used).

This errror has been found in multiple platforms. The 2D amanzi simulation (F-area case) can be found at NERSC: /project/projectdirs/m2398/zexuan/farea/2d_new1.0. The slurm script is run_amanzi.sh with the latest amanzi-1.0 master (/project/projectdirs/m2398/ideas/amanzi/install/cori/devel/bin/amanzi)

Please let me know if you need additional information/description.

**Comment by lipnikov** [comment]
TPLs 0.96.2 updated PFloTran to version b681066 (Sep 20, 2019). 

500 iterations sounds like extreme. A reproducer will certainly help to test more recent PFloTran releases. I am not sure that I am a member of m2398. You may try to share the reproducer on a NERSC machine.

**Comment by zexuanxu** [comment]
You can access the files at /global/cscratch1/sd/zexuanxu/ATS/FAREA_NEW/2d_new1.0

This is a run with the amanzi-1.0 input file (farea_2D_5layer-alq-pflotran_amanzi1.0.xml) you modified for us. 

Thanks!

**Comment by lipnikov** [comment]
Got the files. Thanks. I noticed that you use very tight tolerance on the flow solver. 1e-5 is about 1 Pa error in the pressure. You use 1e-12. 

Also I would recommend to use preconditioning_strategy=linearized_operator which leads to better approximation of Jacobian and faster convergence.

I reproduced the failure with the flow tolerance 1e-7 and the chemistry tolerance 1e-10. I wonder if it is related to Alquimia + PFlotTran problem observed by Sergi.
